feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/swap/executor/ReselectSwapFlowExecutor.kt

```diff
@@ -0,0 +1,44 @@
+package io.novafoundation.nova.feature_assets.presentation.swap.executor
+
+import io.novafoundation.nova.feature_assets.presentation.AssetsRouter
+import io.novafoundation.nova.feature_swap_api.presentation.state.SwapSettingsStateProvider
+import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import kotlinx.coroutines.CoroutineScope
+
+class ReselectSwapFlowExecutorFactory(
+    private val assetsRouter: AssetsRouter,
+    private val swapSettingsStateProvider: SwapSettingsStateProvider,
+    private val chainRegistry: ChainRegistry
+) {
+
+    fun create(selectingDirection: ReselectSwapFlowExecutor.SelectingDirection): ReselectSwapFlowExecutor {
+        return ReselectSwapFlowExecutor(
+            assetsRouter,
+            swapSettingsStateProvider,
+            selectingDirection,
+            chainRegistry
+        )
+    }
+}
```

Comment:
Having a separate factory here seems to be an overhead for me. We can remain only `SwapFlowExecutorFactory` with the same set of benefits